be/src/exec/vectorized/partition/partition_hash_variant.h

```diff
@@ -0,0 +1,284 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "exec/vectorized/partition/partition_hash_map.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks::vectorized {
+
+#define APPLY_FOR_PARTITION_VARIANT_NOT_NULL(M) \
+    M(phase1_uint8)                             \
+    M(phase1_int8)                              \
+    M(phase1_int16)                             \
+    M(phase1_int32)                             \
+    M(phase1_int64)                             \
+    M(phase1_int128)                            \
+    M(phase1_decimal32)                         \
+    M(phase1_decimal64)                         \
+    M(phase1_decimal128)                        \
+    M(phase1_date)                              \
+    M(phase1_timestamp)                         \
+    M(phase1_string)                            \
+    M(phase1_slice)                             \
+    M(phase1_slice_fx4)                         \
+    M(phase1_slice_fx8)                         \
+    M(phase1_slice_fx16)
+
+#define APPLY_FOR_PARTITION_VARIANT_NULL(M) \
+    M(phase1_null_uint8)                    \
+    M(phase1_null_int8)                     \
+    M(phase1_null_int16)                    \
+    M(phase1_null_int32)                    \
+    M(phase1_null_int64)                    \
+    M(phase1_null_int128)                   \
+    M(phase1_null_decimal32)                \
+    M(phase1_null_decimal64)                \
+    M(phase1_null_decimal128)               \
+    M(phase1_null_date)                     \
+    M(phase1_null_timestamp)                \
+    M(phase1_null_string)
+
+#define APPLY_FOR_PARTITION_VARIANT_ALL(M) \
```

Comment:
What's the different from agg_hash_variant.h?

Reply:
Here are the differences:
1. Agg has two stages while partitioner do not, so the the number of fields is different
2. Hash map's entry type is different. In agg the value type is `AggDataPtr`, and in partitioner, the value type is `PartitionChunks`
3. The implementions of hash map are different, they achieve two completely different things